contracts/test/integration/MultiSubnet.t.sol

```diff
@@ -0,0 +1,684 @@
+import {FvmAddress} from "../../src/structs/FvmAddress.sol";
+import {SubnetID, Subnet, IPCAddress, Validator} from "../../src/structs/Subnet.sol";
+import {SubnetIDHelper} from "../../src/lib/SubnetIDHelper.sol";
+import {FvmAddressHelper} from "../../src/lib/FvmAddressHelper.sol";
+import {CrossMsgHelper} from "../../src/lib/CrossMsgHelper.sol";
+import {GatewayDiamond, FEATURE_MULTILEVEL_CROSSMSG} from "../../src/GatewayDiamond.sol";
+import {SubnetActorDiamond} from "../../src/SubnetActorDiamond.sol";
+import {SubnetActorGetterFacet} from "../../src/subnet/SubnetActorGetterFacet.sol";
+import {SubnetActorManagerFacet} from "../../src/subnet/SubnetActorManagerFacet.sol";
+import {SubnetActorCheckpointingFacet} from "../../src/subnet/SubnetActorCheckpointingFacet.sol";
+import {GatewayGetterFacet} from "../../src/gateway/GatewayGetterFacet.sol";
+import {GatewayManagerFacet} from "../../src/gateway/GatewayManagerFacet.sol";
+import {LibGateway} from "../../src/lib/LibGateway.sol";
+import {TopDownFinalityFacet} from "../../src/gateway/router/TopDownFinalityFacet.sol";
+import {CheckpointingFacet} from "../../src/gateway/router/CheckpointingFacet.sol";
+import {XnetMessagingFacet} from "../../src/gateway/router/XnetMessagingFacet.sol";
+import {DiamondCutFacet} from "../../src/diamond/DiamondCutFacet.sol";
+import {GatewayMessengerFacet} from "../../src/gateway/GatewayMessengerFacet.sol";
+import {DiamondLoupeFacet} from "../../src/diamond/DiamondLoupeFacet.sol";
+import {DiamondCutFacet} from "../../src/diamond/DiamondCutFacet.sol";
+import {IntegrationTestBase} from "../IntegrationTestBase.sol";
+import {L2GatewayActorDiamond, L1GatewayActorDiamond} from "../IntegrationTestPresets.sol";
+import {TestUtils, MockIpcContract, MockIpcContractPayable, MockIpcContractFallback} from "../helpers/TestUtils.sol";
+import {FilAddress} from "fevmate/utils/FilAddress.sol";
+import {MerkleTreeHelper} from "../helpers/MerkleTreeHelper.sol";
+
+import {IERC20} from "openzeppelin-contracts/token/ERC20/IERC20.sol";
+import {ERC20PresetFixedSupply} from "../helpers/ERC20PresetFixedSupply.sol";
+import {IERC20Errors} from "openzeppelin-contracts/interfaces/draft-IERC6093.sol";
+
+import "forge-std/console.sol";
+
+contract MultiSubnetTest is Test, IntegrationTestBase {
+    using SubnetIDHelper for SubnetID;
+    using CrossMsgHelper for IpcEnvelope;
+
+    GatewayDiamond public rootGateway;
+    GatewayGetterFacet public rootGatewayGetter;
+    GatewayManagerFacet public rootGatewayManager;
+
+    SubnetActorDiamond public rootNativeSubnetActor;
+    SubnetActorDiamond public rootTokenSubnetActor;
+
+    GatewayDiamond public tokenSubnetGateway;
+    GatewayDiamond public nativeSubnetGateway;
+
+    address[] public nativeSubnetPath;
+    address[] public tokenSubnetPath;
+
+    SubnetID rootSubnetName;
+    SubnetID nativeSubnetName;
+    SubnetID tokenSubnetName;
```

Comment:
For each subnet we have a:
- gateway address
- subnet actor (if not root net)
- name
- path

Instead of repeating all variables, how about we create a struct:

```solidity
struct TestSubnetDefinition {
    address gateway;
    address subnetActor;
    SubnetID id;
    // address[] path << can't we get this from the SubnetID?
}
```

we can then have much better scoping and namespacing:

```
// state variables
TestSubnetDefinition public root;
TestSubnetDefinition public native;
TestSubnetDefinition public token;

// usage
root.gateway.getter()
```